funcx_endpoint/funcx_endpoint/endpoint/endpoint.py

```diff
@@ -225,49 +224,55 @@ def start_endpoint(
+            try:
+                fx_client = Endpoint.get_funcx_client(endpoint_config)
+                reg_info = fx_client.register_endpoint(
+                    endpoint_dir.name,
+                    endpoint_uuid,
+                    metadata=metadata,
+                    multi_tenant=endpoint_config.multi_tenant,
+                )
 
-        except GlobusAPIError as e:
-            if e.http_status < 500:
+            except GlobusAPIError as e:
                 if e.http_status == 409 or e.http_status == 423:
                     # RESOURCE_CONFLICT or RESOURCE_LOCKED
                     log.warning(f"Endpoint registration blocked.  [{e.message}]")
                     exit(os.EX_UNAVAILABLE)
-            raise
+                raise
 
-        except NetworkError as e:
-            # the output of a NetworkError exception is huge and unhelpful, so
-            # it seems better to just stringify it here and get a concise error
-            log.exception(
-                f"Caught exception while attempting endpoint registration: {e}"
-            )
-            log.critical(
-                "funcx-endpoint is unable to reach the funcX service due to a "
-                "NetworkError \n"
-                "Please make sure that the funcX service address you provided is "
-                "reachable \n"
-                "and then attempt restarting the endpoint"
-            )
-            exit(os.EX_TEMPFAIL)
+            except NetworkError as e:
+                # the output of a NetworkError exception is huge and unhelpful, so
+                # it seems better to just stringify it here and get a concise error
+                log.exception(
+                    f"Caught exception while attempting endpoint registration: {e}"
+                )
+                log.critical(
+                    "funcx-endpoint is unable to reach the funcX service due to a "
+                    "NetworkError \n"
+                    "Please make sure that the funcX service address you provided is "
+                    "reachable \n"
+                    "and then attempt restarting the endpoint"
+                )
+                exit(os.EX_TEMPFAIL)
 
-        ret_ep_uuid = reg_info.get("endpoint_id")
-        if ret_ep_uuid != endpoint_uuid:
-            log.error(
-                "Unexpected response from server: mismatched endpoint id."
-                f"\n  Expected: {endpoint_uuid}, received: {ret_ep_uuid}"
-            )
-            exit(os.EX_SOFTWARE)
+            ret_ep_uuid = reg_info.get("endpoint_id")
+            if ret_ep_uuid != endpoint_uuid:
+                log.error(
+                    "Unexpected response from server: mismatched endpoint id."
+                    f"\n  Expected: {endpoint_uuid}, received: {ret_ep_uuid}"
+                )
+                exit(os.EX_SOFTWARE)
+
+        try:
+            endpoint_uuid = reg_info["endpoint_id"]
```

Comment:
Why check only endpoint_id, and not the other fields?  I'd either skip the checking or do all 3 top level fields.  If checking, would it make sense to check in cli._do_start_endpoint at the source instead of a few levels in?

Reply:
Hmm; was thinking that at worst, this would fail within a minute, but that's a valid point.  Let me cogitate on that for a bit.

Reply:
> I'd either skip the checking or do all 3 top level fields.

My initial goal for this was a quick-and-dirty (Q&D) check&nbsp;&mdash;&nbsp;if it at _least_ has the `endpoint_id`, then it's likely good.  Meanwhile, if it fails later on, then let the failure bubble up from the actual usage site.

My thinking was mostly for "who needs to know what?"  Yes, we're passing the `reg_info` along, but it's an opaque dict and so I was going to let the next actor pull out what it knows is there.

That said, I'm not wed to this&nbsp;&mdash;&nbsp;if you disagree with any conviction, note that I've already coded and committed it locally so it can be a simple `git push`.  Let's talk in the morning.

> If checking, would it make sense to check in cli._do_start_endpoint at the source instead of a few levels in?

I'd argue "no."  My reason for doing it here is to treat this as an "all roads lead here" section.  That is, either we get the data from stdin or we get it from upstream.  Either way, we expect it to be the _same data_, and so let's check it in exactly one spot.

Reply:
Definitely the second 'all roads leads here', I had forgotten that there's more than one path.

Reply:
And the quick and dirty check is fine now that I think about it.  Just making sure it's not something completely different.